test/e2e/specs/scaleupdown.go

```diff
@@ -0,0 +1,187 @@
+package specs
+
+import (
+	"context"
+	"flag"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"time"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+
+	"github.com/ghodss/yaml"
+	apiappsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/util/wait"
+
+	internalapi "github.com/openshift/openshift-azure/pkg/api"
+	externalapi "github.com/openshift/openshift-azure/pkg/api/2018-09-30-preview/api"
+	"github.com/openshift/openshift-azure/pkg/util/randomstring"
+	"github.com/openshift/openshift-azure/pkg/util/ready"
+	"github.com/openshift/openshift-azure/test/clients/azure"
+	"github.com/openshift/openshift-azure/test/clients/openshift"
+)
+
+var _ = Describe("Scale Up/Down E2E tests [ScaleUpDown][Real][LongRunning][EveryPR]", func() {
+	const (
+		sampleDeployment = "hello-openshift"
+	)
+	var (
+		azurecli          *azure.Client
+		occli             *openshift.Client
+		scaleUpManifest   = flag.String("scaleUpManifest", "../../test/manifests/normal/scaleup.yaml", "Path to the scale up manifest to send in a partial update request to the RP")
+		scaleDownManifest = flag.String("scaleDownManifest", "../../test/manifests/normal/scaledown.yaml", "Path to the scale down manifest to send in a partial update request to the RP")
+		namespace         string
+	)
+
+	BeforeEach(func() {
+		var err error
+		azurecli, err = azure.NewClientFromEnvironment()
+		Expect(err).NotTo(HaveOccurred())
+		occli, err = openshift.NewEndUserClient()
+		Expect(err).NotTo(HaveOccurred())
+
+		namespace, err = randomstring.RandomString("abcdefghijklmnopqrstuvwxyz0123456789", 5)
+		Expect(err).ToNot(HaveOccurred())
+		namespace = "e2e-test-" + namespace
+		fmt.Fprintln(GinkgoWriter, "Using namespace", namespace)
+		err = occli.CreateProject(namespace)
+		Expect(err).ToNot(HaveOccurred())
+	})
+
+	AfterEach(func() {
+		occli.CleanupProject(namespace)
+	})
+
+	It("should be possible to maintain a healthy cluster after scaling it out and in", func() {
+		ctx := context.Background()
+		ctx = context.WithValue(ctx, internalapi.ContextKeyClientID, os.Getenv("AZURE_CLIENT_ID"))
+		ctx = context.WithValue(ctx, internalapi.ContextKeyClientSecret, os.Getenv("AZURE_CLIENT_SECRET"))
+		ctx = context.WithValue(ctx, internalapi.ContextKeyTenantID, os.Getenv("AZURE_TENANT_ID"))
+
+		By("Fetching the scale up manifest")
+		external, err := parseExternalConfig(*scaleUpManifest)
+		Expect(err).NotTo(HaveOccurred())
+		Expect(external).NotTo(BeNil())
+
+		By("Calling CreateOrUpdate on the rp with the scale up manifest")
+		updated, err := azurecli.OpenShiftManagedClusters.CreateOrUpdate(ctx, os.Getenv("RESOURCEGROUP"), os.Getenv("RESOURCEGROUP"), *external)
```

Comment:
CreateOrUpdate is not enough here because it's an async operation - you need to use the CreateOrUpdateAndWait method I am adding in https://github.com/openshift/openshift-azure/pull/887